### PR TITLE
[BugFix][WeightTransfer] Destroy native HCCL communicators on shutdown

### PR DESCRIPTION
**Pull request by coolhok**
HCCL weight transfer previously dropped its Python communicator reference on shutdown without destroying the native communicator. Reinitialization also replaced the previous communicator without closing it.

Add an idempotent `close()` that synchronizes the communicator's device before calling `hcclCommDestroy`. Call it during shutdown and before reinitialization. Reuse the existing availability flag to prevent repeated destruction and reject collectives on a closed communicator. Initialization, warmup, weight reception, and exception propagation remain unchanged; no restart policy or new exception type is introduced.

### Does this PR introduce _any_ user-facing change?

Yes. Shutdown and reinitialization explicitly release the native HCCL communicator. Calling a collective on an already closed communicator raises `RuntimeError`.

### How was this patch tested?

- 65 unit tests passed in the device-communicator and weight-transfer directories using the repository's CPU/mock UT setup and CI-pinned vLLM revision `84030bbe3d74d99bad477a3d2e37a973ccd8865c` (Python 3.11, PyTorch 2.10.0).
- Regression coverage verifies device synchronization before destruction, repeated/disabled close, close failures, rejection of collectives after close, and engine shutdown/reinitialization.
- Full `bash format.sh ci`, Mypy for the four changed files (`--follow-imports skip --check-untyped-defs`, Python 3.11 target), syntax checks, and `git diff --check` passed.

```bash
pytest -q tests/ut/distributed/device_communicators tests/ut/distributed/weight_transfer
```

A real two-pod/two-NPU Ascend A3 lifecycle smoke test also passed with this exact source revision and the same CI-pinned vLLM source (PyTorch 2.10.0, torch-npu 2.10.0.post2). Each rank completed three communicator initialization/destruction cycles. The test checked weight transfer, all-reduce and broadcast values, explicit-stream operations followed by reinitialization/shutdown, repeated close, and rejection of use after close. Each rank executed exactly three real native destroys; both processes exited successfully. The test used 256-element float32 tensors and a lightweight weight-loading callback.

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

File: vllm_ascend/distributed/device_communicators/pyhccl.py [hunk 1]
```diff
@@ -127,9 +127,21 @@ def __init__(
...
 
+    def close(self) -> None:
+        """Destroy the native communicator at most once."""
+        if not self.available:
+            return
+        self.available = False
+        with torch.npu.device(self.device):
+            # Collectives may have been submitted on an explicit stream.
+            torch.npu.synchronize(self.device)
+            self.hccl.hcclCommDestroy(self.comm)
```
[the remark this answers is not in the document]
Comment (coolhok) [reply vllm_ascend/distributed/device_communicators/pyhccl.py:138]: Thanks for the suggestion. Skipping destruction when synchronization fails is intentional: a failed synchronization does not establish that outstanding communication has completed. Catching the exception and continuing with destruction would suppress the original error and could release resources still in use. The [HCCL documentation](https://www.hiascend.com/document/detail/zh/CANNCommunityEdition/850alpha002/hccl/hcclug/hcclug_000060.html) identifies premature communicator destruction as a possible cause of SDMA errors.
This patch handles normal communicator cleanup. On failure, it propagates the exception and prevents further use or repeated native teardown. Recovery from device or communication failures is outside its scope.

File: vllm_ascend/distributed/weight_transfer/hccl_engine.py [hunk 2]
```diff
@@ -210,7 +212,7 @@ def state_dict_info_iterator():
 
     def shutdown(self) -> None:
         if self.model_update_group is not None:
-            # Clean up the communicator by removing the reference
+            self.model_update_group.close()
             self.model_update_group = None
```
[the remark this answers is not in the document]
Comment (coolhok) [reply vllm_ascend/distributed/weight_transfer/hccl_engine.py:216]: Thanks for the suggestion. close() sets available = False before attempting synchronization or destruction. If it raises, a subsequent shutdown() calls close(), which returns immediately, and then clears the reference. Native synchronization and destruction are therefore not retried.
A finally block would clear the Python reference earlier, but it would not resolve the native cleanup failure. I would prefer to keep the current implementation minimal and let the original exception propagate.

**Review by LCAIZJ — APPROVED** [review]
